test timeout in TestReconnect2

```
=== RUN   TestReconnect2
  2021/09/29 14:37:43 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
  panic: test timed out after 10m0s
  
  goroutine 1247 [running]:
  testing.(*M).startAlarm.func1()
  	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1788 +0xcb
  created by time.goFunc
  	/opt/hostedtoolcache/go/1.17.1/x64/src/time/sleep.go:180 +0x37
  
  goroutine 1 [chan receive, 9 minutes]:
  testing.(*T).Run(0xa4825a0, {0x8842afa, 0xe}, 0x8a1e8d0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1307 +0x359
  testing.runTests.func1(0xa4825a0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1598 +0x5e
  testing.tRunner(0xa4825a0, 0xa44fe70)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1259 +0xeb
  testing.runTests(0xa496c00, {0x9001ff0, 0x2, 0x2}, {0xc04d3d03eb186370, 0x8bb3986c44, 0x900ab00})
  	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1596 +0x39a
  testing.(*M).Run(0xa4aa190)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1504 +0x581
  main.main()
  	_testmain.go:45 +0x136
  
  goroutine 6 [select, 9 minutes]:
  github.com/ipfs/go-log/writer.(*MirrorWriter).logRoutine(0xa5125a0)
  	/home/runner/go/pkg/mod/github.com/ipfs/go-log@v1.0.5/writer/writer.go:71 +0xe3
  created by github.com/ipfs/go-log/writer.NewMirrorWriter
  	/home/runner/go/pkg/mod/github.com/ipfs/go-log@v1.0.5/writer/writer.go:36 +0xbb
  
  goroutine 30 [semacquire, 9 minutes]:
  sync.runtime_Semacquire(0xb57ffe8)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/sema.go:56 +0x36
  sync.(*WaitGroup).Wait(0xb57ffe0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/sync/waitgroup.go:130 +0x87
  github.com/libp2p/go-libp2p/p2p/test/reconnects.SubtestConnSendDisc(0xa482690, {0xa9b1f94, 0x2, 0x2})
  	/home/runner/work/go-libp2p/go-libp2p/p2p/test/reconnects/reconnect_test.go:202 +0x32f
  github.com/libp2p/go-libp2p/p2p/test/reconnects.TestReconnect2(0xa482690)
  	/home/runner/work/go-libp2p/go-libp2p/p2p/test/reconnects/reconnect_test.go:120 +0x2d2
  testing.tRunner(0xa482690, 0x8a1e8d0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1259 +0xeb
  created by testing.(*T).Run
  	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1306 +0x33a
  
  goroutine 12 [select, 9 minutes]:
  github.com/libp2p/go-libp2p-peerstore/pstoremem.(*memoryAddrBook).background(0xa5c8900)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.2.8/pstoremem/addr_book.go:93 +0xeb
  created by github.com/libp2p/go-libp2p-peerstore/pstoremem.NewAddrBook
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.2.8/pstoremem/addr_book.go:83 +0x1b1
  
  goroutine 13 [select, 5 minutes]:
  github.com/libp2p/go-libp2p-swarm.(*DialBackoff).background(0xa519f30, {0x8adbd68, 0xa89a990})
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_dial.go:123 +0xdf
  created by github.com/libp2p/go-libp2p-swarm.(*DialBackoff).init
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_dial.go:116 +0x70
  
  goroutine 14 [IO wait, 9 minutes]:
  internal/poll.runtime_pollWait(0xe7b814a8, 0x72)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/netpoll.go:229 +0x8e
  internal/poll.(*pollDesc).wait(0xa510424, 0x72, 0x0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:84 +0x37
  internal/poll.(*pollDesc).waitRead(...)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:89
  internal/poll.(*FD).Accept(0xa510410)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_unix.go:402 +0x215
  net.(*netFD).accept(0xa510410)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/fd_unix.go:173 +0x27
  net.(*TCPListener).accept(0xa8c4010)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/tcpsock_posix.go:140 +0x27
  net.(*TCPListener).Accept(0xa8c4010)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/tcpsock.go:262 +0x37
  github.com/multiformats/go-multiaddr/net.(*maListener).Accept(0xa8c4050)
  	/home/runner/go/pkg/mod/github.com/multiformats/go-multiaddr@v0.4.0/net/net.go:243 +0x2a
  github.com/libp2p/go-tcp-transport.(*tcpListener).Accept(0xa8c4070)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-tcp-transport@v0.3.0/tcp.go:83 +0x2a
  github.com/libp2p/go-tcp-transport.(*tracingListener).Accept(0xa40d810)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-tcp-transport@v0.3.0/metrics.go:261 +0x26
  github.com/libp2p/go-libp2p-transport-upgrader.(*listener).handleIncoming(0xa89aae0)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.5.0/listener.go:73 +0xbf
  created by github.com/libp2p/go-libp2p-transport-upgrader.(*Upgrader).UpgradeListener
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.5.0/upgrader.go:48 +0x1a7
  
  goroutine 15 [chan receive, 9 minutes]:
  github.com/libp2p/go-libp2p-transport-upgrader.(*listener).Accept(0xa89aae0)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.5.0/listener.go:151 +0x4a
  github.com/libp2p/go-libp2p-swarm.(*Swarm).AddListenAddr.func2({0x8adbf08, 0xa89aae0}, 0xa519e60, {0x8aea0b4, 0xa8c4040}, {0x8aea0b4, 0xa496470})
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_listen.go:94 +0x13d
  created by github.com/libp2p/go-libp2p-swarm.(*Swarm).AddListenAddr
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_listen.go:79 +0x232
  
  goroutine 16 [select]:
  github.com/libp2p/go-libp2p-quic-transport.(*reuse).gc(0xa89c630)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/reuse.go:91 +0x12c
  created by github.com/libp2p/go-libp2p-quic-transport.newReuse
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/reuse.go:69 +0xf9
  
  goroutine 33 [select]:
  github.com/libp2p/go-libp2p-quic-transport.(*reuse).gc(0xa89c648)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/reuse.go:91 +0x12c
  created by github.com/libp2p/go-libp2p-quic-transport.newReuse
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/reuse.go:69 +0xf9
  
  goroutine 34 [IO wait, 9 minutes]:
  internal/poll.runtime_pollWait(0xe7b81410, 0x72)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/netpoll.go:229 +0x8e
  internal/poll.(*pollDesc).wait(0xa510514, 0x72, 0x0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:84 +0x37
  internal/poll.(*pollDesc).waitRead(...)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:89
  internal/poll.(*FD).RawRead(0xa510500, 0xb6e91e0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_unix.go:554 +0x131
  net.(*rawConn).Read(0xa40d8a8, 0xb6e91e0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/rawconn.go:43 +0x3f
  golang.org/x/net/internal/socket.(*Conn).recvMsgs(0xa8c47b0, {0xa51e580, 0x8, 0x8}, 0x0)
  	/home/runner/go/pkg/mod/golang.org/x/net@v0.0.0-20210813160813-60bc85c4be6d/internal/socket/rawconn_mmsg.go:32 +0x1e3
  golang.org/x/net/internal/socket.(*Conn).RecvMsgs(...)
  	/home/runner/go/pkg/mod/golang.org/x/net@v0.0.0-20210813160813-60bc85c4be6d/internal/socket/socket.go:267
  golang.org/x/net/ipv4.(*payloadHandler).ReadBatch(0xa89ac98, {0xa51e580, 0x8, 0x8}, 0x0)
  	/home/runner/go/pkg/mod/golang.org/x/net@v0.0.0-20210813160813-60bc85c4be6d/ipv4/batch.go:80 +0x63
  github.com/lucas-clemente/quic-go.(*oobConn).ReadPacket(0xa5b5140)
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/conn_oob.go:151 +0x6b1
  github.com/lucas-clemente/quic-go.(*packetHandlerMap).listen(0xa576620)
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/packet_handler_map.go:329 +0x59
  created by github.com/lucas-clemente/quic-go.newPacketHandlerMap
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/packet_handler_map.go:145 +0x281
  
  goroutine 35 [select, 9 minutes]:
  github.com/lucas-clemente/quic-go.(*baseServer).run(0xa8c6060)
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/server.go:230 +0xde
  created by github.com/lucas-clemente/quic-go.listen
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/server.go:216 +0x5fe
  
  goroutine 36 [select, 9 minutes]:
  github.com/lucas-clemente/quic-go.(*baseServer).accept(0xa8c6060, {0x8adbd88, 0xa4320c0})
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/server.go:268 +0x99
  github.com/lucas-clemente/quic-go.(*baseServer).Accept(0xa8c6060, {0x8adbd88, 0xa4320c0})
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/server.go:264 +0x31
  github.com/libp2p/go-libp2p-quic-transport.(*listener).Accept(0xa89ad20)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/listener.go:64 +0x3e
  github.com/libp2p/go-libp2p-swarm.(*Swarm).AddListenAddr.func2({0x8adbee8, 0xa89ad20}, 0xa519e60, {0x8aea0b4, 0xa8c4810}, {0x8aea0b4, 0xa8c4770})
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_listen.go:94 +0x13d
  created by github.com/libp2p/go-libp2p-swarm.(*Swarm).AddListenAddr
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_listen.go:79 +0x232
  
  goroutine 38 [select, 9 minutes]:
  github.com/libp2p/go-libp2p/p2p/protocol/identify.(*ObservedAddrManager).worker(0xa4718c0)
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/obsaddr.go:258 +0x1a8
  created by github.com/libp2p/go-libp2p/p2p/protocol/identify.NewObservedAddrManager
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/obsaddr.go:158 +0x42a
  
  goroutine 39 [select, 9 minutes]:
  github.com/libp2p/go-libp2p/p2p/protocol/identify.(*IDService).loop(0xa507300)
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/id.go:203 +0x505
  created by github.com/libp2p/go-libp2p/p2p/protocol/identify.NewIDService
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/id.go:150 +0x2cd
  
  goroutine 40 [select, 9 minutes]:
  github.com/libp2p/go-libp2p-peerstore/pstoremem.(*memoryAddrBook).background(0xa5c9200)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.2.8/pstoremem/addr_book.go:93 +0xeb
  created by github.com/libp2p/go-libp2p-peerstore/pstoremem.NewAddrBook
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-peerstore@v0.2.8/pstoremem/addr_book.go:83 +0x1b1
  
  goroutine 41 [select, 5 minutes]:
  github.com/libp2p/go-libp2p-swarm.(*DialBackoff).background(0xa8d81f0, {0x8adbd68, 0xa89b470})
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_dial.go:123 +0xdf
  created by github.com/libp2p/go-libp2p-swarm.(*DialBackoff).init
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_dial.go:116 +0x70
  
  goroutine 42 [IO wait, 9 minutes]:
  internal/poll.runtime_pollWait(0xe7b81378, 0x72)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/netpoll.go:229 +0x8e
  internal/poll.(*pollDesc).wait(0xa510654, 0x72, 0x0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:84 +0x37
  internal/poll.(*pollDesc).waitRead(...)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:89
  internal/poll.(*FD).Accept(0xa510640)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_unix.go:402 +0x215
  net.(*netFD).accept(0xa510640)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/fd_unix.go:173 +0x27
  net.(*TCPListener).accept(0xa8c4f20)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/tcpsock_posix.go:140 +0x27
  net.(*TCPListener).Accept(0xa8c4f20)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/tcpsock.go:262 +0x37
  github.com/multiformats/go-multiaddr/net.(*maListener).Accept(0xa8c4f60)
  	/home/runner/go/pkg/mod/github.com/multiformats/go-multiaddr@v0.4.0/net/net.go:243 +0x2a
  github.com/libp2p/go-tcp-transport.(*tcpListener).Accept(0xa8c4f80)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-tcp-transport@v0.3.0/tcp.go:83 +0x2a
  github.com/libp2p/go-tcp-transport.(*tracingListener).Accept(0xa40dad8)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-tcp-transport@v0.3.0/metrics.go:261 +0x26
  github.com/libp2p/go-libp2p-transport-upgrader.(*listener).handleIncoming(0xa89b5c0)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.5.0/listener.go:73 +0xbf
  created by github.com/libp2p/go-libp2p-transport-upgrader.(*Upgrader).UpgradeListener
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.5.0/upgrader.go:48 +0x1a7
  
  goroutine 43 [chan receive, 9 minutes]:
  github.com/libp2p/go-libp2p-transport-upgrader.(*listener).Accept(0xa89b5c0)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-transport-upgrader@v0.5.0/listener.go:151 +0x4a
  github.com/libp2p/go-libp2p-swarm.(*Swarm).AddListenAddr.func2({0x8adbf08, 0xa89b5c0}, 0xa8d8120, {0x8aea0b4, 0xa8c4f50}, {0x8aea0b4, 0xa496470})
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_listen.go:94 +0x13d
  created by github.com/libp2p/go-libp2p-swarm.(*Swarm).AddListenAddr
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_listen.go:79 +0x232
  
  goroutine 44 [select]:
  github.com/libp2p/go-libp2p-quic-transport.(*reuse).gc(0xa89d068)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/reuse.go:91 +0x12c
  created by github.com/libp2p/go-libp2p-quic-transport.newReuse
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/reuse.go:69 +0xf9
  
  goroutine 45 [select]:
  github.com/libp2p/go-libp2p-quic-transport.(*reuse).gc(0xa89d080)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/reuse.go:91 +0x12c
  created by github.com/libp2p/go-libp2p-quic-transport.newReuse
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/reuse.go:69 +0xf9
  
  goroutine 46 [IO wait, 9 minutes]:
  internal/poll.runtime_pollWait(0xe7b812e0, 0x72)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/netpoll.go:229 +0x8e
  internal/poll.(*pollDesc).wait(0xa510744, 0x72, 0x0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:84 +0x37
  internal/poll.(*pollDesc).waitRead(...)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:89
  internal/poll.(*FD).RawRead(0xa510730, 0xb7cb040)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_unix.go:554 +0x131
  net.(*rawConn).Read(0xa40db68, 0xb7cb040)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/rawconn.go:43 +0x3f
  golang.org/x/net/internal/socket.(*Conn).recvMsgs(0xa8c5690, {0xa51e9a0, 0x8, 0x8}, 0x0)
  	/home/runner/go/pkg/mod/golang.org/x/net@v0.0.0-20210813160813-60bc85c4be6d/internal/socket/rawconn_mmsg.go:32 +0x1e3
  golang.org/x/net/internal/socket.(*Conn).RecvMsgs(...)
  	/home/runner/go/pkg/mod/golang.org/x/net@v0.0.0-20210813160813-60bc85c4be6d/internal/socket/socket.go:267
  golang.org/x/net/ipv4.(*payloadHandler).ReadBatch(0xa89b778, {0xa51e9a0, 0x8, 0x8}, 0x0)
  	/home/runner/go/pkg/mod/golang.org/x/net@v0.0.0-20210813160813-60bc85c4be6d/ipv4/batch.go:80 +0x63
  github.com/lucas-clemente/quic-go.(*oobConn).ReadPacket(0xa8f2a00)
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/conn_oob.go:151 +0x6b1
  github.com/lucas-clemente/quic-go.(*packetHandlerMap).listen(0xa577030)
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/packet_handler_map.go:329 +0x59
  created by github.com/lucas-clemente/quic-go.newPacketHandlerMap
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/packet_handler_map.go:145 +0x281
  
  goroutine 47 [select, 9 minutes]:
  github.com/lucas-clemente/quic-go.(*baseServer).run(0xa8c6120)
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/server.go:230 +0xde
  created by github.com/lucas-clemente/quic-go.listen
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/server.go:216 +0x5fe
  
  goroutine 48 [select, 9 minutes]:
  github.com/lucas-clemente/quic-go.(*baseServer).accept(0xa8c6120, {0x8adbd88, 0xa4320c0})
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/server.go:268 +0x99
  github.com/lucas-clemente/quic-go.(*baseServer).Accept(0xa8c6120, {0x8adbd88, 0xa4320c0})
  	/home/runner/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.23.0/server.go:264 +0x31
  github.com/libp2p/go-libp2p-quic-transport.(*listener).Accept(0xa89b800)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-quic-transport@v0.13.0/listener.go:64 +0x3e
  github.com/libp2p/go-libp2p-swarm.(*Swarm).AddListenAddr.func2({0x8adbee8, 0xa89b800}, 0xa8d8120, {0x8aea0b4, 0xa8c56f0}, {0x8aea0b4, 0xa8c5660})
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_listen.go:94 +0x13d
  created by github.com/libp2p/go-libp2p-swarm.(*Swarm).AddListenAddr
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_listen.go:79 +0x232
  
  goroutine 52 [select, 9 minutes]:
  github.com/libp2p/go-libp2p/p2p/protocol/identify.(*ObservedAddrManager).worker(0xa471c20)
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/obsaddr.go:258 +0x1a8
  created by github.com/libp2p/go-libp2p/p2p/protocol/identify.NewObservedAddrManager
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/obsaddr.go:158 +0x42a
  
  goroutine 53 [select, 9 minutes]:
  github.com/libp2p/go-libp2p/p2p/protocol/identify.(*IDService).loop(0xa9a6080)
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/id.go:203 +0x505
  created by github.com/libp2p/go-libp2p/p2p/protocol/identify.NewIDService
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/id.go:150 +0x2cd
  
  goroutine 988 [IO wait]:
  internal/poll.runtime_pollWait(0xe7b811b0, 0x72)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/netpoll.go:229 +0x8e
  internal/poll.(*pollDesc).wait(0xb47f3c4, 0x72, 0x0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:84 +0x37
  internal/poll.(*pollDesc).waitRead(...)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:89
  internal/poll.(*FD).Read(0xb47f3b0, {0xb57fb20, 0xc, 0xc})
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_unix.go:167 +0x1d4
  net.(*netFD).Read(0xb47f3b0, {0xb57fb20, 0xc, 0xc})
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/fd_posix.go:56 +0x3f
  net.(*conn).Read(0xb2f55d8, {0xb57fb20, 0xc, 0xc})
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/net.go:183 +0x4f
  io.ReadAtLeast({0xe7aff1a0, 0xb7d2d80}, {0xb57fb20, 0xc, 0xc}, 0xc)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/io/io.go:328 +0x97
  io.ReadFull(...)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/io/io.go:347
  github.com/libp2p/go-yamux/v2.(*Session).recvLoop(0xab6a870)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:583 +0xa3
  github.com/libp2p/go-yamux/v2.(*Session).recv(0xab6a870)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:562 +0x21
  created by github.com/libp2p/go-yamux/v2.newSession
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:132 +0x4b0
  
  goroutine 100 [chan receive, 9 minutes]:
  github.com/libp2p/go-flow-metrics.(*sweeper).run(0x900ac40)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-flow-metrics@v0.0.3/sweeper.go:40 +0x4a
  created by github.com/libp2p/go-flow-metrics.(*sweeper).start
  	/home/runner/go/pkg/mod/github.com/libp2p/go-flow-metrics@v0.0.3/sweeper.go:36 +0x6c
  
  goroutine 996 [select, 9 minutes]:
  github.com/libp2p/go-yamux/v2.(*Session).AcceptStream(0xab6a870)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:232 +0xcf
  github.com/libp2p/go-libp2p-yamux.(*conn).AcceptStream(0xab6a870)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-yamux@v0.5.4/conn.go:35 +0x21
  github.com/libp2p/go-libp2p-swarm.(*Conn).start.func1(0xa8c6b40)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_conn.go:107 +0xcb
  created by github.com/libp2p/go-libp2p-swarm.(*Conn).start
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_conn.go:102 +0x33
  
  goroutine 1019 [select, 9 minutes]:
  github.com/libp2p/go-libp2p/p2p/protocol/identify.(*peerHandler).loop(0xb800d40, {0x8adbd68, 0xb6ef2c0}, 0xb6fe7c0)
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/peer_loop.go:90 +0xcb
  created by github.com/libp2p/go-libp2p/p2p/protocol/identify.(*peerHandler).start
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/peer_loop.go:72 +0x8d
  
  goroutine 1076 [chan receive, 9 minutes]:
  github.com/libp2p/go-libp2p/p2p/test/reconnects.SubtestConnSendDisc.func1(0xb57ffe0, 0xb2f5360, {0x8aeaa4c, 0xb7da3a0}, 0xb59ff00, 0xb5894b0, 0x1)
  	/home/runner/work/go-libp2p/go-libp2p/p2p/test/reconnects/reconnect_test.go:192 +0x210
  created by github.com/libp2p/go-libp2p/p2p/test/reconnects.SubtestConnSendDisc
  	/home/runner/work/go-libp2p/go-libp2p/p2p/test/reconnects/reconnect_test.go:182 +0x189
  
  goroutine 1011 [select, 9 minutes]:
  github.com/libp2p/go-yamux/v2.(*Session).AcceptStream(0xab6a900)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:232 +0xcf
  github.com/libp2p/go-libp2p-yamux.(*conn).AcceptStream(0xab6a900)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-yamux@v0.5.4/conn.go:35 +0x21
  github.com/libp2p/go-libp2p-swarm.(*Conn).start.func1(0xa8c6ba0)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_conn.go:107 +0xcb
  created by github.com/libp2p/go-libp2p-swarm.(*Conn).start
  	/home/runner/go/pkg/mod/github.com/libp2p/go-libp2p-swarm@v0.6.0/swarm_conn.go:102 +0x33
  
  goroutine 1088 [chan send, 9 minutes]:
  github.com/libp2p/go-libp2p/p2p/test/reconnects.newSender.func1.1({0x8aeaa4c, 0xb7da3a0}, {0xb7d6b80, 0xb7d6bc0, 0xb7d6c00, 0xb7d6c40, 0xb7d6c80})
  	/home/runner/work/go-libp2p/go-libp2p/p2p/test/reconnects/reconnect_test.go:63 +0x39
  github.com/libp2p/go-libp2p/p2p/test/reconnects.newSender.func1({0x8aeaa4c, 0xb7da3a0})
  	/home/runner/work/go-libp2p/go-libp2p/p2p/test/reconnects/reconnect_test.go:86 +0x759
  created by github.com/libp2p/go-libp2p/p2p/test/reconnects.SubtestConnSendDisc.func1
  	/home/runner/work/go-libp2p/go-libp2p/p2p/test/reconnects/reconnect_test.go:185 +0x85
  
  goroutine 1002 [select]:
  github.com/libp2p/go-yamux/v2.(*Session).sendLoop(0xab6a900)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:506 +0x1b6
  github.com/libp2p/go-yamux/v2.(*Session).send(0xab6a900)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:452 +0x21
  created by github.com/libp2p/go-yamux/v2.newSession
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:133 +0x4ce
  
  goroutine 1034 [select, 9 minutes]:
  github.com/libp2p/go-libp2p/p2p/protocol/identify.(*peerHandler).loop(0xb801140, {0x8adbd68, 0xb6ef3e0}, 0xb6feaa0)
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/peer_loop.go:90 +0xcb
  created by github.com/libp2p/go-libp2p/p2p/protocol/identify.(*peerHandler).start
  	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/identify/peer_loop.go:72 +0x8d
  
  goroutine 989 [select]:
  github.com/libp2p/go-yamux/v2.(*Session).sendLoop(0xab6a870)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:506 +0x1b6
  github.com/libp2p/go-yamux/v2.(*Session).send(0xab6a870)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:452 +0x21
  created by github.com/libp2p/go-yamux/v2.newSession
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:133 +0x4ce
  
  goroutine 1001 [IO wait]:
  internal/poll.runtime_pollWait(0xe7b81248, 0x72)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/netpoll.go:229 +0x8e
  internal/poll.(*pollDesc).wait(0xb47f004, 0x72, 0x0)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:84 +0x37
  internal/poll.(*pollDesc).waitRead(...)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_poll_runtime.go:89
  internal/poll.(*FD).Read(0xb47eff0, {0xb57fbc0, 0xc, 0xc})
  	/opt/hostedtoolcache/go/1.17.1/x64/src/internal/poll/fd_unix.go:167 +0x1d4
  net.(*netFD).Read(0xb47eff0, {0xb57fbc0, 0xc, 0xc})
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/fd_posix.go:56 +0x3f
  net.(*conn).Read(0xb2f5700, {0xb57fbc0, 0xc, 0xc})
  	/opt/hostedtoolcache/go/1.17.1/x64/src/net/net.go:183 +0x4f
  io.ReadAtLeast({0xe7aff1a0, 0xb6eec30}, {0xb57fbc0, 0xc, 0xc}, 0xc)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/io/io.go:328 +0x97
  io.ReadFull(...)
  	/opt/hostedtoolcache/go/1.17.1/x64/src/io/io.go:347
  github.com/libp2p/go-yamux/v2.(*Session).recvLoop(0xab6a900)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:583 +0xa3
  github.com/libp2p/go-yamux/v2.(*Session).recv(0xab6a900)
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:562 +0x21
  created by github.com/libp2p/go-yamux/v2.newSession
  	/home/runner/go/pkg/mod/github.com/libp2p/go-yamux/v2@v2.2.0/session.go:132 +0x4b0
  FAIL	github.com/libp2p/go-libp2p/p2p/test/reconnects	600.029s

Minimal reproducer:
```go
func TestConnectHang(t *testing.T) {
	h1, err := bhost.NewHost(swarmt.GenSwarm(t, swarmt.OptDisableQUIC), nil)
	require.NoError(t, err)
	h2, err := bhost.NewHost(swarmt.GenSwarm(t, swarmt.OptDisableQUIC), nil)
	require.NoError(t, err)

	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()
	require.NoError(t, h1.Connect(ctx, peer.AddrInfo{ID: h2.ID(), Addrs: h2.Addrs()}))
	require.Len(t, h1.Network().Conns(), 1)
	require.Len(t, h2.Network().Conns(), 1)

	done := make(chan struct{})
	go func() {
		h2.SetStreamHandler(protocol.TestingID, func(str network.Stream) {
			defer close(done)
			io.ReadAll(str)
		})
	}()
	str, err := h1.NewStream(ctx, h2.ID(), protocol.TestingID)
	require.NoError(t, err)
	_, err = str.Write([]byte("foobar"))
	require.NoError(t, err)
	require.NoError(t, str.CloseWrite())

	fmt.Println(h1.Network().Conns()[0])
	require.NoError(t, h1.Network().Conns()[0].Close())
	require.Eventually(t, func() bool { return len(h1.Network().Conns()) == 0 }, 10*time.Second, 10*time.Millisecond)
	require.Eventually(t, func() bool { return len(h2.Network().Conns()) == 0 }, 10*time.Second, 10*time.Millisecond)
}
```

output:
```
=== RUN   TestConnectHang
<swarm.Conn[*tcp.TcpTransport] /ip4/127.0.0.1/tcp/57237 (12D3KooWB1SFhGLrzyQKptrFxkC7KYY9pHCgQ9BXu4YacPgKaNo2) <-> /ip4/127.0.0.1/tcp/57238 (12D3KooWFrC4Ffr2NwCQaVWjyiD66mkyZ4952g2MahnE8MW3XkED)>
    reconnect_test.go:138: 
                Error Trace:    reconnect_test.go:138
                Error:          Condition never satisfied
                Test:           TestConnectHang
--- FAIL: TestConnectHang (10.02s)
```

pcap: [conn-hanging.pcapng.gz](https://github.com/libp2p/go-libp2p/files/7725650/conn-hanging.pcapng.gz)
<img width="1283" alt="Screen Shot 2021-12-16 at 12 35 41" src="https://user-images.githubusercontent.com/1478487/146336591-75437f70-8612-4565-b019-d2750e735ebd.png">

The interesting thing is that both TCP RSTs are sent, so both connections should be closed. From a go routine dump it looks like the `ReadFull` in https://github.com/libp2p/go-yamux/blob/959e04413929c5c82f518e992954e42076676d05/session.go#L584 is hanging.

--- a/p2p/test/reconnects/reconnect_test.go
+++ b/p2p/test/reconnects/reconnect_test.go
@@ -8,15 +8,17 @@ import (
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/require"
+	bhost "github.com/libp2p/go-libp2p/p2p/host/basic"
 
-	u "github.com/ipfs/go-ipfs-util"
-	logging "github.com/ipfs/go-log/v2"
 	"github.com/libp2p/go-libp2p-core/host"
 	"github.com/libp2p/go-libp2p-core/network"
 	"github.com/libp2p/go-libp2p-core/protocol"
+
 	swarmt "github.com/libp2p/go-libp2p-swarm/testing"
-	bhost "github.com/libp2p/go-libp2p/p2p/host/basic"
+
+	u "github.com/ipfs/go-ipfs-util"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/stretchr/testify/require"
 )
 
 var log = logging.Logger("reconnect")
@@ -102,9 +104,11 @@ func newSender() (chan sendChans, func(s network.Stream)) {
 
 // TestReconnect tests whether hosts are able to disconnect and reconnect.
 func TestReconnect2(t *testing.T) {
-	h1, err := bhost.NewHost(swarmt.GenSwarm(t), nil)
+	// TCP RST handling is flaky in OSX, see https://github.com/golang/go/issues/50254.
+	// We can avoid this by using QUIC in this test.
+	h1, err := bhost.NewHost(swarmt.GenSwarm(t, swarmt.OptDisableTCP), nil)
 	require.NoError(t, err)
-	h2, err := bhost.NewHost(swarmt.GenSwarm(t), nil)
+	h2, err := bhost.NewHost(swarmt.GenSwarm(t, swarmt.OptDisableTCP), nil)
 	require.NoError(t, err)
 	hosts := []host.Host{h1, h2}
 
@@ -117,7 +121,7 @@ func TestReconnect2(t *testing.T) {
 	}
 	for i := 0; i < rounds; i++ {
 		log.Debugf("TestReconnect: %d/%d\n", i, rounds)
-		SubtestConnSendDisc(t, hosts)
+		subtestConnSendDisc(t, hosts)
 	}
 }
 
@@ -126,7 +130,9 @@ func TestReconnect5(t *testing.T) {
 	const num = 5
 	hosts := make([]host.Host, 0, num)
 	for i := 0; i < num; i++ {
-		h, err := bhost.NewHost(swarmt.GenSwarm(t), nil)
+		// TCP RST handling is flaky in OSX, see https://github.com/golang/go/issues/50254.
+		// We can avoid this by using QUIC in this test.
+		h, err := bhost.NewHost(swarmt.GenSwarm(t, swarmt.OptDisableTCP), nil)
 		require.NoError(t, err)
 		h.SetStreamHandler(protocol.TestingID, EchoStreamHandler)
 		hosts = append(hosts, h)
@@ -138,12 +144,11 @@ func TestReconnect5(t *testing.T) {
 	}
 	for i := 0; i < rounds; i++ {
 		log.Debugf("TestReconnect: %d/%d\n", i, rounds)
-		SubtestConnSendDisc(t, hosts)
+		subtestConnSendDisc(t, hosts)
 	}
 }
 
-func SubtestConnSendDisc(t *testing.T, hosts []host.Host) {
-
+func subtestConnSendDisc(t *testing.T, hosts []host.Host) {
 	ctx := context.Background()
 	numStreams := 3 * len(hosts)
 	numMsgs := 10

